entry/main.cc

```diff
@@ -1,9 +1,49 @@
 #include <iostream>
+#include "crow.h"
+//#include "crow_all.h"
+#include <ctime>
+#include <iostream>
+#include <unistd.h>
+#include <openssl/sha.h>
+#include "auth.h"
 
 #include "engine/matching.h"
 
 int main() {
     get_pnl_snapshots({ "me", "mine" });
+    
+    crow::SimpleApp app; //define your crow application
+    auth a;
+
+    //define your endpoint at the root directory
+    CROW_ROUTE(app, "/")([&a](const crow::request& req){
+        
+        if (a.validate_credentials(req))
+            return crow::response(200);
+        else
+            return crow::response(401);
+    });
+
+    CROW_ROUTE(app, "/getcredentials")([&a](){
+        return a.generate_credentials();
+    });
+
+    CROW_ROUTE(app, "/refreshcredentials")([&a](const crow::request& req){
+        
+        if (a.validate_refresh_token(req)){
```

Comment:
just return a `crow::response` from validate_refresh_token?

I think if validate_refresh_token was static and returned that, this could be:
`CROW_ROUTE(app, "/refreshcredentials", Auth::validate_refresh_token);`